frontend/pulse3d/src/components/interactiveAnalysis/InteractiveAnalysisModal.js

```diff
@@ -241,39 +482,69 @@ export default function InteractiveWaveformModal({
     }
   }, []);
 
+  // Update baseData anytime the remove dups checkbox state changes, except when set during load of existing changes
   useEffect(() => {
+    if (loadStatus === LOAD_STATUSES.LOADING_EXISTING) {
+      return;
+    }
     if (removeDupsChecked) {
-      const currentPeaksValleysCopy = JSON.parse(JSON.stringify(editablePeaksValleys));
-      // remove duplicate peaks and valleys per well
-      for (const well in currentPeaksValleysCopy) {
-        currentPeaksValleysCopy[well] = removeWellSpecificDuplicates(well);
+      const baseDataCopy = deepCopy(baseData);
+      for (const well in baseDataCopy) {
+        const { duplicateFeatureIndices } = customAnalysisSettings[well];
+        baseDataCopy[well][0] = removeDups(baseDataCopy[well][0], duplicateFeatureIndices.peaks);
+        baseDataCopy[well][1] = removeDups(baseDataCopy[well][1], duplicateFeatureIndices.valleys);
       }
-      setBaseData(currentPeaksValleysCopy);
-    } else {
+      setBaseData(baseDataCopy);
```

Comment:
So is baseData always the source of truth for initial analaysis data regardless of if the remove duplicates feature is ever checked or it's only getting set when remove duplicates is interacted with?

Reply:
It's a copy of the original peaks and valleys but can have dups removed. It's the "base" data in that it's what resetting will return to